### ISSUE TITLE
Update HS Ontology API to use Neo4j 5.x

### ISSUE THREAD

**Issue by shirey** [x-atlas-consortia/hs-ontology-api#70]
Create a new version of the HS Ontology API that will work against Neo4j 5.x. For now create a "neo4jv5" branch for this work and don't merge into main or dev yet. This branch will need to be released in coordination with updates to 

  - Instead of using the released version of UBKG API, base this on the Neo4j 5 branch of UBKG [here](https://github.com/x-atlas-consortia/ubkg-api/tree/neo4jv5),  eventually we will switch back to using the UBKG API main/released version once it's neo4jv5 branch has been released.
  - Update the Neo python library/driver to the matching 5.x version
  - Test all endpoints/neo4j queries to make sure they still work in Neo4j 5, list any broken endpoints for future repair in a new issue

**Comment by ChuckKollar** [comment]
Draft PR: https://github.com/x-atlas-consortia/hs-ontology-api/pull/79

**Comment by ChuckKollar** [comment]
This work now also includes the following issues: https://github.com/x-atlas-consortia/ubkg-api/issues/93, and https://github.com/x-atlas-consortia/hs-ontology-api/issues/80

**Comment by ChuckKollar** [comment]
Bill Shirey Jan 26, 2024
Yes,

Alan we need to run the v5 branches in ubkg-neo4j, ubkg-api and hs-ontology-api until all is ready for release, I think..

Joe and Charles I like the idea of comparing the existing “v4” calls against the in-progress “v5"- need to make sure the data is the same/compatible on both when we do this, though- so Alan as long as you have a v5 of the HuBMAP-SenNet DB that matches PROD, seems like we’d be in shape-- if not, I know Jonathan wants the other stuff that you’re working on out first…

**Comment by yuanzhou** [comment]
Thanks to @ChuckKollar for creating the script https://github.com/x-atlas-consortia/hs-ontology-api/blob/neo4jv5/src/compare_responses.py to compare the v5 output against the current v4 on DEV/PROD. 

Attached is the diff file: [new_cypher_diff.log](https://github.com/x-atlas-consortia/hs-ontology-api/files/14144781/new_cypher_diff.log)

@AlanSimmons will need to rerun the comparison when he reviews the changes later, before we can merge the following to PRs and test on DEV:
- https://github.com/x-atlas-consortia/hs-ontology-api/pull/79 and PR 
- https://github.com/x-atlas-consortia/ubkg-api/pull/92
